File: mobile/android/app/src/main/java/com/ap/ap/model/Customer.kt
package com.ap.ap.model

data class Customer(
    val id: String,
    val name: String,
    val company: String,
    val address: String,
    val latitude: Double?,
    val longitude: Double?,
    val geocodeStatus: GeocodeStatus,
    val visitStatus: VisitStatus,
    val visitedAt: String? = null,
)

enum class VisitStatus {
    VISITED,
    UNVISITED,
}

enum class GeocodeStatus {
    SUCCESS,
    FAILED,
    PENDING,
}

File: mobile/h5/src/types/customer.ts
export type VisitStatus = "VISITED" | "UNVISITED";
export type GeocodeStatus = "SUCCESS" | "FAILED" | "PENDING";

export interface Customer {
  id: string;
  name: string;
  company: string;
  address: string;
  latitude: number | null;
  longitude: number | null;
  geocodeStatus: GeocodeStatus;
  visitStatus: VisitStatus;
  visitedAt?: string | null;
}

export interface CustomerQuery {
  status?: "visited" | "unvisited";
  radiusKm?: number;
  centerLat?: number;
  centerLng?: number;
}
